# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

In [58]:
USE LaborstatisticsDB
SELECT TOP 50*
FROM annual_2016
ORDER BY series_id

(50 rows affected)

Total execution time: 00:00:00.108

id series_id year period value footnote_codes original_file 8755 CEU0000000001 2016 M13 144306 NULL ce.data.0.AllCESSeries.csv 22220 CEU0000000001 2016 M13 144306 NULL ce.data.00a.TotalNonfarm.Employment.csv 8756 CEU0000000010 2016 M13 71492 NULL ce.data.0.AllCESSeries.csv 22221 CEU0000000010 2016 M13 71492 NULL ce.data.00a.TotalNonfarm.Employment.csv 6281 CEU0500000001 2016 M13 122083 NULL ce.data.05a.TotalPrivate.Employment.csv 8757 CEU0500000001 2016 M13 122083 NULL ce.data.0.AllCESSeries.csv 7954 CEU0500000002 2016 M13 34.4 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8758 CEU0500000002 2016 M13 34.4 NULL ce.data.0.AllCESSeries.csv 7955 CEU0500000003 2016 M13 25.65 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8759 CEU0500000003 2016 M13 25.65 NULL ce.data.0.AllCESSeries.csv 6282 CEU0500000006 2016 M13 100531 NULL ce.data.05a.TotalPrivate.Employment.csv 8760 CEU0500000006 2016 M13 100531 NULL ce.data.0.AllCESSeries.csv 5721 CEU0500000007 2016 M13 33.6 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8761 CEU0500000007 2016 M13 33.6 NULL ce.data.0.AllCESSeries.csv 5722 CEU0500000008 2016 M13 21.54 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8762 CEU0500000008 2016 M13 21.54 NULL ce.data.0.AllCESSeries.csv 6283 CEU0500000010 2016 M13 58747 NULL ce.data.05a.TotalPrivate.Employment.csv 8763 CEU0500000010 2016 M13 58747 NULL ce.data.0.AllCESSeries.csv 7956 CEU0500000011 2016 M13 881.55 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8764 CEU0500000011 2016 M13 881.55 NULL ce.data.0.AllCESSeries.csv 3808 CEU0500000012 2016 M13 367.3 NULL ce.data.02b.AllRealEarningsAE.csv 7957 CEU0500000012 2016 M13 367.3 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8765 CEU0500000012 2016 M13 367.3 NULL ce.data.0.AllCESSeries.csv 3809 CEU0500000013 2016 M13 10.69 NULL ce.data.02b.AllRealEarningsAE.csv 7958 CEU0500000013 2016 M13 10.69 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8766 CEU0500000013 2016 M13 10.69 NULL ce.data.0.AllCESSeries.csv 7959 CEU0500000016 2016 M13 105.3 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8767 CEU0500000016 2016 M13 105.3 NULL ce.data.0.AllCESSeries.csv 7960 CEU0500000017 2016 M13 129.1 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8768 CEU0500000017 2016 M13 129.1 NULL ce.data.0.AllCESSeries.csv 5723 CEU0500000030 2016 M13 723.69 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8769 CEU0500000030 2016 M13 723.69 NULL ce.data.0.AllCESSeries.csv 5724 CEU0500000031 2016 M13 309.17 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8770 CEU0500000031 2016 M13 309.17 NULL ce.data.0.AllCESSeries.csv 26206 CEU0500000031 2016 M13 309.17 NULL ce.data.03c.AllRealEarningsPE.csv 5725 CEU0500000032 2016 M13 9.2 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8771 CEU0500000032 2016 M13 9.2 NULL ce.data.0.AllCESSeries.csv 26207 CEU0500000032 2016 M13 9.2 NULL ce.data.03c.AllRealEarningsPE.csv 5726 CEU0500000034 2016 M13 112.5 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8772 CEU0500000034 2016 M13 112.5 NULL ce.data.0.AllCESSeries.csv 5727 CEU0500000035 2016 M13 162 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8773 CEU0500000035 2016 M13 162 NULL ce.data.0.AllCESSeries.csv 7961 CEU0500000056 2016 M13 4195743 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8774 CEU0500000056 2016 M13 4195743 NULL ce.data.0.AllCESSeries.csv 7962 CEU0500000057 2016 M13 107621623 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8775 CEU0500000057 2016 M13 107621623 NULL ce.data.0.AllCESSeries.csv 5728 CEU0500000081 2016 M13 3377192 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv 8776 CEU0500000081 2016 M13 3377192 NULL ce.data.0.AllCESSeries.csv 5729 CEU0500000082 2016 M13 72752850 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndE

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees?        
 ANS : The data type for women employees is 10.</span>

In [48]:
USE LaborstatisticsDB
SELECT TOP 50* FROM datatype
WHERE data_type_text = 'women employees'

(1 row affected)

Total execution time: 00:00:00.083

data_type_code data_type_text 10 WOMEN EMPLOYEES

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?
 ANS: SeriesID : </span> <span style="font-size: 13.3333px; white-space: pre-wrap;">CES5552211010</span> and CEU5552211010

In [50]:
SELECT supersector_code, supersector_name 
FROM supersector
WHERE supersector_name LIKE '%financial%'

(1 row affected)

Total execution time: 00:00:00.074

supersector_code supersector_name 55 Financial activities

In [53]:
SELECT industry_code, industry_name
FROM industry
WHERE industry_name LIKE '%commercial banking%'

(1 row affected)

Total execution time: 00:00:00.076

industry_code industry_name 55522110 Commercial banking

In [60]:
SELECT series_id FROM series 
WHERE data_type_code LIKE '10'
AND supersector_code LIKE '55'
AND industry_code LIKE '55522110'

(2 rows affected)

Total execution time: 00:00:00.078

series_id CES5552211010 CEU5552211010

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [61]:
USE LaborStatisticsDB
SELECT TOP 10 a16.id, a16.series_id, a16.year, a16.period, a16.value, a16.footnote_codes, a16.original_file FROM annual_2016 AS a16
LEFT JOIN series AS s
ON a16.series_id = s.series_id 

(10 rows affected)

Total execution time: 00:00:00.223

id series_id year period value footnote_codes original_file 0 CEU5500000007 2016 M13 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CEU5500000008 2016 M13 26.11 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CEU5500000030 2016 M13 962.73 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CEU5500000031 2016 M13 411.29 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CEU5500000032 2016 M13 11.15 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 5 CEU5500000034 2016 M13 111.6 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 6 CEU5500000035 2016 M13 179.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 7 CEU5500000081 2016 M13 236997 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 8 CEU5500000082 2016 M13 6189003 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 9 CEU5552200007 2016 M13 37.3 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [19]:
SELECT TOP 10 * FROM series 
LEFT JOIN datatype 
ON series.data_type_code = datatype.data_type_code

(10 rows affected)

Total execution time: 00:00:00.081

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [20]:
USE LaborStatisticsDB
 SELECT TOP 10 * FROM series
LEFT JOIN industry
ON series.industry_code = industry.industry_code

(10 rows affected)

Total execution time: 00:00:00.089

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please. ANS : 2340612 employees in all industries reported in 2016.</span>

In [17]:
USE LaborStatisticsDB 
-- select series_id,series_title from series
SELECT sum(value) AS 'Total Employees' FROM annual_2016
WHERE series_id LIKE '%01' --All employees
--seriesID and data_type_code are interlinked 

(1 row affected)

Total execution time: 00:00:00.126

Total Employees 2340612.2000000016

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; ANS : The total women employees in all industries in the year 2016 are 1125490.</span>

In [21]:
-- select series_id,series_title from series
SELECT SUM(value) AS 'Women employees' FROM LaborStatisticsDB.dbo.annual_2016
WHERE series_id LIKE '%10' --Women employees
     

(1 row affected)

Total execution time: 00:00:00.303

Women employees 1125489.6000000008

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp; ANS : Total employees in the production and nonsupervisory industries are 1263650.</span>

In [20]:
-- select series_id,series_title from series
SELECT sum(value) AS 'Total employees' FROM LaborStatisticsDB.dbo.annual_2016
WHERE series_id LIKE '%06' --Production and nonsupervisory employees

(1 row affected)

Total execution time: 00:00:00.115

Total employees 1263649.6000000008

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?ANS : The average weekly hours worked by production and nonsupervisory employess across all industries is 36 hours/week.</span>

In [22]:
-- select series_id,series_title from series
SELECT AVG(value) AS 'Average weeekly hours' FROM LaborStatisticsDB.dbo.january_2017 
WHERE series_id LIKE '%07' --Average weekly hours of production and nonsupervisory employees

(1 row affected)

Total execution time: 00:00:00.517

Average weeekly hours 36.05862068965521

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please.</span>

In [23]:
-- SELECT series_id,series_title from series
SELECT SUM(j17.value) AS weekly, industry_name FROM january_2017 as j17
JOIN industry as i
ON j17.id = i.id
WHERE series_id LIKE '%30'  -- series_id 30 is Average weekly earnings of production and nonsupervisory employees
AND industry_name IS NOT NULL
GROUP BY industry_name
ORDER BY weekly DESC
     

(74 rows affected)

Total execution time: 00:00:00.107

weekly industry_name 1847.47 Monetary authorities - central bank 1773.7 Motor vehicle bodies and trailers 1655.98 General freight trucking 1614.25 Support activities for transportation 1581.46 Machine shops and threaded products 1560 Farm machinery and equipment 1558.66 Pet and pet supplies stores 1491.35 Other nonferrous metal production 1466.04 Transit and ground passenger transportation 1427.77 Fuel dealers 1412.75 Metal valves 1395.45 Plate work and fabricated structural products 1268.78 Special tools 1248.21 Children's and infants' clothing stores 1233.74 Information 1230.53 Wireless telecommunications carriers (except satellite) 1218.97 Motion picture and video production 1207.71 Wiring devices 1204.1 Printed circuit assemblies 1194.9 Hazardous waste treatment and disposal 1186.69 Electrical equipment and appliances 1179.62 All other specialty trade contractors 1174.24 Plumbing equipment 1133.94 Power generation and supply 1120.75 Warehouse clubs and supercenters 1102.45 Computer and peripheral equipment 1096.96 Direct life and health insurance carriers 1057.6 Other insurance-related activities 1014.93 Surgical appliances and supplies 1010.79 Residential building equipment contractors 1007.12 Carpet and upholstery cleaning services 1000.51 Computer and software 997.9 Druggists' goods 994.19 Other nondepository credit intermediation 987.37 Hobby 969.36 Goods-producing 965.68 Motor vehicle seating and interior trim 951.26 Securities brokerage 946.91 Soaps 936.03 Millwork 924.71 Ship building and repairing 880.88 Petroleum 855.75 Bituminous coal and lignite surface mining 851.24 Miscellaneous durable goods 823.55 Colleges and universities 804.86 Recreational vehicle dealers 795.87 Electronic markets and agents and brokers 792.1 Computer facilities management services 776.42 Other home furnishings stores 773.91 Residential remodelers 773.15 Converted paper products 770.41 Other nonmetallic mineral mining 763.73 Land subdivision 759.62 Other building material dealers 756.45 Environmental consulting services 744.75 Poured concrete structure contractors 738.53 Apparel 726.64 Fiber 726.13 Miniwarehouse and self-storage unit operators 721.09 Title abstract and settlement offices 712.92 Animal food 685.08 Employment placement agencies and executive search services 679.08 Public relations agencies 667.14 Other basic organic chemicals 640.04 Machinery and equipment rental and leasing 635.01 Rental and leasing services 631.9 Bakeries and tortilla manufacturing 630.13 Frozen specialty food 623.66 Furniture and furnishings 614.17 Telemarketing bureaus 607 Soft drinks 600.22 Investigation and security services 596.55 Food (health) supplement stores 592.88 Painting and wall covering contractors

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest? </span> ANS : Highest average weekly hours of production and nonsupervisory employees are from 'waste collection' industry and the lowest is 'offices of lawyers'.

In [42]:
USE LaborStatisticsDB
SELECT TOP 1 j17.value, i.industry_name FROM january_2017 as j17
JOIN industry as i 
ON j17.id = i.id 
WHERE j17.series_id LIKE '%07'  --series_id 07 is Average weekly hours of production and nonsupervisory employees
ORDER BY j17.value DESC 

(1 row affected)

Total execution time: 00:00:00.244

value industry_name 42 Waste collection

In [43]:
SELECT TOP 1 j17.value, i.industry_name FROM january_2017 as j17
JOIN industry as i 
ON j17.id = i.id 
WHERE j17.series_id LIKE '%07' -- series_id 07 is Average weekly hours of production and nonsupervisory employees
ORDER BY j17.value 

(1 row affected)

Total execution time: 00:00:00.201

value industry_name 30.4 Offices of lawyers

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest? ANS : The highest weekly payroll for production and nonsupervisory employees is '</span><span style="font-size: 13.3333px; white-space: pre-wrap;">Monetary authorities - central bank' with weekly payroll of 1848$ and the lowest is 'Painting and wall covering contractors' with weekly payroll of $593.</span>

In [15]:
USE LaborStatisticsDB
SELECT TOP 3 j17.value, i.industry_name FROM january_2017 as j17
JOIN industry as i 
ON j17.id = i.id 
WHERE j17.series_id LIKE '%30'  -- series_id 30 is Average weekly earnings of production and nonsupervisory employees
ORDER BY j17.value DESC

(3 rows affected)

Total execution time: 00:00:00.283

value industry_name 1847.47 Monetary authorities - central bank 1773.7 Motor vehicle bodies and trailers 1655.98 General freight trucking

In [31]:
USE LaborStatisticsDB
SELECT TOP 1 j17.value, i.industry_name FROM january_2017 as j17
JOIN industry as i 
ON j17.id = i.id 
WHERE j17.series_id LIKE '%30' 
ORDER BY j17.value 
     

(1 row affected)

Total execution time: 00:00:00.289

value industry_name 592.88 Painting and wall covering contractors

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [59]:
SELECT TOP 10 j17.series_id, j17.value AS "weekly payroll", i.industry_code, i.industry_name FROM january_2017 AS j17
LEFT JOIN industry as i 
ON j17.id = i.id 
WHERE i.industry_name IS NOT NULL
AND j17.value >
(
    SELECT AVG(a16.value) 
    FROM annual_2016 as a16
    WHERE a16.series_id LIKE '%82' --both data_type_code and series_id are same -Aggregate weekly payrolls of production and nonsupervisory employees
) 


(10 rows affected)

Total execution time: 00:00:00.130

series_id weekly payroll industry_code industry_name CES5500000082 6328968 10211000 Oil and gas extraction CES5552210082 971796 20236115 New single-family general contractors CES5552300082 1044224 31332100 Forging and stamping CES5552400082 2347328 31334400 Semiconductors and electronic components CES5552410082 1477799 31334515 Electricity and signal testing instruments CES5552420082 872398 31336413 Other aircraft parts and equipment CES5553100082 862987 32311400 Fruit and vegetable preserving and specialty CEU5500000082 6467426 41424500 Farm product raw materials CEU5552200082 1767137 41424920 Books and periodicals CEU5552210082 1004069 42441110 New car dealers

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [14]:

SELECT top 1 a16.year, a16.period, a16.value AS "average earnings"
FROM annual_2016 AS a16 
WHERE a16.series_id LIKE '%30' 
UNION 
SELECT top 1 j17.year, j17.period, j17.value
FROM january_2017 AS j17 
WHERE j17.series_id LIKE '%30'
ORDER BY "average earnings" DESC

(2 rows affected)

Total execution time: 00:00:00.322

year period average earnings 2017 M01 969.36 2016 M13 962.73

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

The year 2017 has better production and supervisory employees fare than the year 2016.

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

Banking (Monetary authories-Central bank) followed by Motor vehicle bodies and trailers and General freight trucking has the better fare for production and nonsupervisory employees.

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

Had lot of confusions with the value columns it would have been given separately with values like money,person count,etc. There were many unneccessary columns which made it difficult to understand the tables.